cvs -f log -N -r1.39 kimageeffect.cpp

RCS file: /home/kde/kdelibs/kdefx/kimageeffect.cpp,v
Working file: kimageeffect.cpp
head: 1.39
branch:
locks: strict
access list:
keyword substitution: kv
total revisions: 40;	selected revisions: 1
description:
----------------------------
revision 1.39
date: 2002/12/02 08:47:13;  author: staikos;  state: Exp;  lines: +14 -1
Make blendOnLower() work on big endian systems.
=============================================================================
cvs -f diff -bp -u -r1.38 -r1.39 kimageeffect.cpp
Index: kimageeffect.cpp
===================================================================
RCS file: /home/kde/kdelibs/kdefx/kimageeffect.cpp,v
retrieving revision 1.38
retrieving revision 1.39
diff -b -p -u -r1.38 -r1.39
--- kdefx/kimageeffect.cpp	2002/07/31 22:52:45	1.38
+++ kdefx/kimageeffect.cpp	2002/12/02 08:47:13	1.39
@@ -27,7 +27,7 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE PO
 
 */
 
-// $Id: kdelibs-3.1-kimageeffect.patch,v 1.2 2002/12/03 13:06:31 hannes Exp $
+// $Id: kdelibs-3.1-kimageeffect.patch,v 1.2 2002/12/03 13:06:31 hannes Exp $
 
 #include <math.h>
 
@@ -1958,11 +1958,16 @@ bool KImageEffect::blendOnLower(
     --b; --i;
     do
     {
+#ifndef WORDS_BIGENDIAN
       while ( !(a=*i) && k>0 )
+#else
+      while ( !(a=*(i-3)) && k>0 )
+#endif
       {
         i-=4; b-=4; k--;
       };
 
+#ifndef WORDS_BIGENDIAN
       --i; --b;
       *b += ( ((*i - *b) * a) >> 8 );
       --i; --b;
@@ -1970,6 +1975,14 @@ bool KImageEffect::blendOnLower(
       --i; --b;
       *b += ( ((*i - *b) * a) >> 8 );
       --i; --b;
+#else
+      *b += ( ((*i - *b) * a) >> 8 );
+      --i; --b;
+      *b += ( ((*i - *b) * a) >> 8 );
+      --i; --b;
+      *b += ( ((*i - *b) * a) >> 8 );
+      i -= 2; b -= 2;
+#endif
     } while (k--);
   }
 
